src/queue.c

```diff
@@ -0,0 +1,101 @@
+#include "queue.h"
+
+#include <stdio.h>
+#include <stdlib.h>
+#define TRUE 1
+#define FALSE 0
+
+struct node
+{
+    long start_time;
+    long wait_time;
+    struct node *next;
+};
+typedef struct node node;
+
+struct queue
+{
+    int count;
+    node *front;
+    node *rear;
+};
+typedef struct queue queue;
+
+void initialize(queue *q)
+{
+    q->count = 0;
+    q->front = NULL;
+    q->rear = NULL;
+}
+
+void display(node *head)
+{
+    if(head == NULL)
+    {
+        printf("NULL\n");
+    }
+    else
+    {
+        printf("%d\n", head -> wait_time);
+        display(head->next);
+    }
+}
+
+int isempty(queue *q)
+{
+    return (q->rear == NULL);
+}
+
+// Returns the time at which the enqueued flow will leave the switch and enqueues the flow
+long enqueue(queue *q, long start_time, long wait_time)
+{
+        node *tmp;
+        tmp = malloc(sizeof(node));
```

Comment:
Add an `assert(tmp != NULL)` after this line to make sure we only proceed if malloc worked fine. (Assert will exit the program if tmp is NULL)